src/NUnitTestAdapterTests/TestExecutionTests.cs

```diff
@@ -54,16 +54,20 @@ public void LoadMockassembly()
             MockAssemblyPath = Path.Combine(TestContext.CurrentContext.TestDirectory, "mock-assembly.dll");
 
             // Sanity check to be sure we have the correct version of mock-assembly.dll
-            Assert.That(MockAssembly.Tests, Is.EqualTo(34),
+            Assert.That(MockAssembly.TestsAtRuntime , Is.EqualTo(MockAssembly.Tests),
                 "The reference to mock-assembly.dll appears to be the wrong version");
             new List<TestCase>();
-            testResults = new List<TestResult>();
             testLog = new FakeFrameworkHandle();
 
             // Load the NUnit mock-assembly.dll once for this test, saving
             // the list of test cases sent to the discovery sink
             executor = ((ITestExecutor) new NUnit3TestExecutor());
             executor.RunTests(new[] { MockAssemblyPath }, Context, testLog);
+
+            testResults = testLog.Events
+               .Where(e => e.EventType == FakeFrameworkHandle.EventType.RecordResult)
+               .Select(e => e.TestResult)
+               .ToList();
```

Comment:
testResults was always an empty list in previous implementation, that is how to ExplicitTestDoesNotShowUpInResults managed to pass even if the test was recorded with outcome None